aesara/tensor/basic_opt.py

```diff
@@ -1610,7 +1610,18 @@ def dimshuffled_alloc(i):
                         cond.append(eq(i_shp, cmp_shp))
                 if cond:
                     assert_op_in = assert_op(assert_op_in, *cond)
-            new_i.append(i.owner.inputs[0])
+            alloc_input = i.owner.inputs[0]
+            if alloc_input.ndim != i.owner.inputs[0].ndim:
+                # The `Alloc` can add dimension to the value
```

Comment:
This looks like a very odd thing. Should we just intercept `Alloc` calls that add new dimensions and return `dimshuffles` instead?